File: Chap1-Events/1-6OverrideEventMethodDemo/button.cpp
#include "button.h"
#include <QEvent>
#include <QDebug>

Button::Button(QWidget *parent) : QPushButton(parent)
{

}

bool Button::event(QEvent *event)
{
    if( (event->type() == QEvent::MouseButtonPress)
                || (event->type() == QEvent::MouseButtonDblClick)){
        qDebug() << "Button : mouse press or doubleclick detected";
      //  return true;
    }
    return  QPushButton::event(event);
}

File: Chap1-Events/1-6OverrideEventMethodDemo/button.py
from typing import Optional
from PySide6.QtWidgets import QPushButton
from PySide6.QtCore import QEvent

class Button(QPushButton):
    def __init__(self, parent: Optional[QPushButton] = None) -> None:
        super().__init__(parent)
    
    def event(self, event: QEvent) -> bool:
        if (event.type() == QEvent.MouseButtonPress or 
            event.type() == QEvent.MouseButtonDblClick):
            print("Button: mouse press or doubleclick detected")
            # return True
        
        return super().event(event)